File: pdlearning/src/web-apps/src/libraries/domain-api/src/course/dtos/search-course-users-request.ts
export interface ISearchCourseUserRequest {
  searchText: string;
  forCourse?: ISearchUsersQueryForCourseInfo;
  skipCount: number;
  maxResultCount: number;
}

export interface ISearchUsersQueryForCourseInfo {
  courseId: string;
  followCourseTargetParticipant: boolean;
}

File: pdlearning/src/microservices/course-microservice/Api/Application/RequestDtos/CourseRequest/SearchCourseUsersRequest.cs
using System;
using Thunder.Platform.Core.Application.Dtos;

#pragma warning disable SA1402 // File may only contain a single type
namespace Microservice.Course.Application.RequestDtos
{
    public class SearchCourseUsersRequest : PagedResultRequestDto
    {
        public string SearchText { get; set; }

        public SearchUsersQueryForCourseInfo ForCourse { get; set; }
    }

    public class SearchUsersQueryForCourseInfo
    {
        public Guid CourseId { get; set; }

        public bool FollowCourseTargetParticipant { get; set; }
    }
}
#pragma warning restore SA1402 // File may only contain a single type
